src/Elliptic/DiscontinuousGalerkin/NumericalFluxes/FirstOrderInternalPenalty.hpp

```diff
@@ -0,0 +1,388 @@
+    };
+    EXPAND_PACK_LEFT_TO_RIGHT(helper(FieldTags{}, AuxiliaryFieldTags{},
+                                     normal_dot_auxiliary_field_fluxes));
+  }
+
+  // This function combines local and remote data to the numerical fluxes.
+  // The numerical fluxes as not-null pointers are the first arguments. The
+  // other arguments are the packaged types for the interior side followed by
+  // the packaged types for the exterior side.
+  void operator()(
+      const gsl::not_null<db::item_type<::Tags::NormalDotNumericalFlux<
+          FieldTags>>*>... numerical_flux_for_fields,
+      const gsl::not_null<db::item_type<::Tags::NormalDotNumericalFlux<
+          AuxiliaryFieldTags>>*>... numerical_flux_for_auxiliary_fields,
+      const db::item_type<::Tags::NormalDotFlux<
+          AuxiliaryFieldTags>>&... normal_dot_auxiliary_flux_interiors,
+      const db::item_type<NormalDotDivFlux<
+          AuxiliaryFieldTags>>&... normal_dot_div_auxiliary_flux_interiors,
+      const db::item_type<NormalDotNormalDotFlux<
+          AuxiliaryFieldTags>>&... ndot_ndot_aux_flux_interiors,
+      const db::item_type<::Tags::NormalDotFlux<
+          AuxiliaryFieldTags>>&... minus_normal_dot_auxiliary_flux_exteriors,
+      const db::item_type<NormalDotDivFlux<
+          AuxiliaryFieldTags>>&... minus_normal_dot_div_aux_flux_exteriors,
+      const db::item_type<NormalDotDivFlux<
+          AuxiliaryFieldTags>>&... ndot_ndot_aux_flux_exteriors) const
+      noexcept {
+    // Need polynomial degress and element size to compute this dynamically
+    const double penalty = penalty_parameter_;
+
+    const auto helper = [&penalty](
+        const auto numerical_flux_for_field,
+        const auto numerical_flux_for_auxiliary_field,
+        const auto normal_dot_auxiliary_flux_interior,
+        const auto normal_dot_div_auxiliary_flux_interior,
+        const auto ndot_ndot_aux_flux_interior,
+        const auto minus_normal_dot_auxiliary_flux_exterior,
+        const auto minus_normal_dot_div_aux_flux_exterior,
+        const auto ndot_ndot_aux_flux_exterior) noexcept {
+      for (auto it = numerical_flux_for_auxiliary_field->begin();
+           it != numerical_flux_for_auxiliary_field->end(); it++) {
+        const auto index =
+            numerical_flux_for_auxiliary_field->get_tensor_index(it);
+        *it = 0.5 * (normal_dot_auxiliary_flux_interior.get(index) -
+                     minus_normal_dot_auxiliary_flux_exterior.get(index));
+      }
+      for (auto it = numerical_flux_for_field->begin();
+           it != numerical_flux_for_field->end(); it++) {
+        const auto index = numerical_flux_for_field->get_tensor_index(it);
+        *it = 0.5 * (normal_dot_div_auxiliary_flux_interior.get(index) -
+                     minus_normal_dot_div_aux_flux_exterior.get(index)) -
+              penalty * (ndot_ndot_aux_flux_interior.get(index) -
+                         ndot_ndot_aux_flux_exterior.get(index));
```

Comment:
What happened to the F^i_u in each term in Eq. (2) in the help text, which you are implementing here? I'm also unclear why there are aux flux terms in the coefficient of penalty, when your help text has only F^i_u, not F^i_v in those terms.

Reply:
- The F^i_u are included here, just not in this variable name. They are the first thing that's computed in `package_data`.
- Good catch with the F^i_v in the penalty term! They were missing in the docs. I added them and also changed the tensor-product notation to indices. Please let me know if this makes it more clear now.